persistence/sql/update_process_execution_from_rpc.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2023 XDBLab Organization
+// SPDX-License-Identifier: BUSL-1.1
+
+package sql
+
+import (
+	"context"
+	"github.com/xdblab/xdb/common/log/tag"
+	"github.com/xdblab/xdb/extensions"
+	"github.com/xdblab/xdb/persistence/data_models"
+)
+
+func (p sqlProcessStoreImpl) UpdateProcessExecutionFromRpc(ctx context.Context, request data_models.UpdateProcessExecutionFromRpcRequest) (
+	*data_models.UpdateProcessExecutionFromRpcResponse, error) {
+	tx, err := p.session.StartTransaction(ctx, defaultTxOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := p.doUpdateProcessExecutionFromRpcTx(ctx, tx, request)
+
+	if err != nil || resp.FailAtUpdatingGlobalAttributes {
+		err2 := tx.Rollback()
+		if err2 != nil {
+			p.logger.Error("error on rollback transaction", tag.Error(err2))
+		}
+	} else {
+		err = tx.Commit()
+		if err != nil {
+			p.logger.Error("error on committing transaction", tag.Error(err))
+			return nil, err
+		}
+	}
+
+	return resp, err
+}
+
+func (p sqlProcessStoreImpl) doUpdateProcessExecutionFromRpcTx(
+	ctx context.Context, tx extensions.SQLTransaction, request data_models.UpdateProcessExecutionFromRpcRequest,
+) (*data_models.UpdateProcessExecutionFromRpcResponse, error) {
+	hasNewImmediateTask := false
+
+	// lock process execution row first
+	prcRow, err := tx.SelectProcessExecutionForUpdate(ctx, request.ProcessExecutionId)
+	if err != nil {
+		return nil, err
+	}
+
+	// skip the writing operations on a closed process
```

Comment:
Instead of silently skip this update, it's better to return an error(maybe conflict error) for this case. 

users should be informed that the update is rejected otherwise they won't have the correct expectation for what has happened. (this is the behavior in iWF today without any extra code because Cadence/Temporal will throw error when sending signals to a closed workflow execution)

Reply:
Will return StatusNotFound